### PR TITLE
Swaps: Gas fee range to use for native asset maxSwappableAmount buffer

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-1615
Fixes APP-1616

### CODE REVIEW

File: src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx [hunk 1]
```diff
@@ -84,55 +87,59 @@ const getHasEnoughFundsForGas = (quote: Quote, gasFee: string, nativeNetworkAsse
...
+
+    const isEstimateOutsideRange =
+      gasFeeRange.current &&
+      (lessThanWorklet(nativeGasFee, gasFeeRange.current[0]) || greaterThanWorklet(nativeGasFee, gasFeeRange.current[1]));
+
+    // If the gas fee range hasn't been set or the estimated fee is outside the range, calculate the range based on the gas fee
+    if (nativeGasFee && (!gasFeeRange.current || isEstimateOutsideRange)) {
+      const lowerBound = toFixedWorklet(mulWorklet(nativeGasFee, LOWER_BOUND_FACTOR), assetToSell.decimals);
+      const upperBound = toFixedWorklet(mulWorklet(nativeGasFee, UPPER_BOUND_FACTOR), assetToSell.decimals);
+      gasFeeRange.current = [lowerBound, upperBound];
```
Comment (greg-schrammel) [inline src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx:122]: do we need this if assetToSell is not native? could do this work after the isNativeAsset condition?

Comment (benisgold) [reply src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx]: No but I think it's good to keep it like this, so if the user switches from a non native to the native asset then there will already be a gas fee range stored and we can immediately update maxSwappableAmount

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 2]
```diff
@@ -559,6 +561,24 @@ export function useSwapInputsController({
...
+  // update the input amount & quote if swapping max amount & maxSwappableAmount changes
+  useAnimatedReaction(
+    () => maxSwappableAmount.value,
+    maxSwappableAmount => {
+      const isSwappingMaxBalance = internalSelectedInputAsset.value && inputMethod.value === 'slider' && percentageToSwap.value >= 1;
+      if (maxSwappableAmount && isSwappingMaxBalance) {
+        inputValues.modify(prev => {
+          return {
+            ...prev,
+            inputAmount: +maxSwappableAmount,
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:573]: note: we'll make a separate change so we don't have to do this +

**Review by jinchung — APPROVED** [review]
🌮 note that this is review of code only, unable to run / test locally while i'm on a plane

**Pull request merged**